rosbag2_transport/include/rosbag2_transport/qos.hpp

```diff
@@ -40,11 +42,15 @@ namespace rosbag2_transport
 class Rosbag2QoS : public rclcpp::QoS
 {
 public:
+  ROSBAG2_TRANSPORT_PUBLIC
```

Comment:
I think you might be able to achieve this with just `class ROSBAG2_TRANSPORT_PUBLIC Rosbag2QoS` instead of marking all methods. Feels tidier to me